### DISCUSSION TITLE
Sideband immediately quits at launch with no error message

### DISCUSSION THREAD

**Discussion by xa4hf8** [markqvist/Reticulum#634] [Help & Questions] [answered]
**M1 Mac**
Mac OS Sequoia 15.1.1
Python 3.12.1
Sideband 1.2

[admin console log](https://github.com/user-attachments/files/18139056/Sideband.silently.exits.at.launch.txt)

[assistant analysis](https://chatgpt.com/share/675e561c-c970-8012-b554-e61f990eff28)

**Comment by markqvist** [comment]
Is this running from `Sideband.app` provided in the dmg file, or installed via `pip`?

**Comment by markqvist** [comment, marked as answer]
If it is running from the app package, could you try launching it via the terminal, and see if any additional error messages are output? Open a terminal, `cd` to the directory that the `Sideband.app` package is stored in, and run:

`./Sideband.app/Contents/MacOS/Sideband`

There may be additional error messages there, that can shed some light on what is going on.

**Reply by xa4hf8** [reply]
> `./Sideband.app/Contents/MacOS/Sideband`
> 
> There may be additional error messages there, that can shed some light on what is going on.

I should have thought of that before reporting: turns out that it does not like the space in the path name.
Sorry for the trouble.

 ```
% ./Sideband
/Applications/Conferencing/Wireless off-grid/Reticulum/Sideband.app/Contents/Resources/script: line 10: cd: /Applications/Conferencing
off-grid/Reticulum/Sideband.app/Contents/Resources: No such file or directory
/Applications/Conferencing/Wireless off-grid/Reticulum/Sideband.app/Contents/Resources/script: line 13: pushd: /Applications/Conferencing
off-grid/Reticulum/Sideband.app/Contents/Resources/venv/bin: No such file or directory
/Applications/Conferencing/Wireless off-grid/Reticulum/Sideband.app/Contents/Resources/script: line 15: activate: No such file or directory
/Applications/Conferencing/Wireless off-grid/Reticulum/Sideband.app/Contents/Resources/script: line 50: exec: python: not found

```

**Reply by liamcottle** [reply]
Can confirm it also crashes for me with similar error messages when launching `Sideband.app` by either double clicking, or running via CLI when a parent directory contains a space.

I'd say it's probably best to move it to `/Applications/Sideband.app`.

**Reply by markqvist** [reply]
Aha! So that's what's happening. Good catch. It occurs because the process needs to change working directory into the virtual python environment included in the app, but for some reason can't handle spaces in the name then.

I'll see if I can implement a workaround in the next release.

**Comment by markqvist** [comment]
Ok, to investigate, I just launched a fresh macOS Sequoia instance running on Mac M1 hardware via EC2, and downloaded and installed the latest Sideband release. Everything runs as intended.

It could be that you simply have errors in your Reticulum configuration file, that prevents startup. Or a bit more unlikely, that you've upgraded from a *very* old version of Sideband, and the new version can't properly migrate database version and configuration files from that old version.

In that case, you can try moving the existing Sideband configuration folder (`~/.config/sideband`) to a backup location, which will make Sideband re-initialise config and database files on startup. If that works, you will have to manually import your old encryption key into the new config folder to use your existing address.

Moving this to the forums, since it currently does not appear to be an issue.

**Comment by liamcottle** [comment]
Working for me on:

- Sideband v1.2.0
- macOS Sonoma 14.2.1 (23C71)
- 2019 MacBook Pro, Intel


I don't have an M1 Mac, nor Mac OS Sequoia 15.x install to test with

**Reply by markqvist** [reply]
Thanks!

**Comment by briandinsmore** [comment]
Just a note, I ended up having to use sudo to flash an Rnode on a new Mac instance and it appears to have created the .config directory in the home directory with the permissions of "root" and "staff" group.  This would cause the app to open for around half a second before the window disappeared with the icon.  No error message that I was seeing in the terminal console when I manually opened the app and it crashed.  CHOWN'ing the .conf directory fixed it
<img width="424" alt="Screenshot 2025-05-30 at 7 52 36 PM" src="https://github.com/user-attachments/assets/e7b98c1d-4ccd-46cc-9e73-8f16ac9cf057" />

**Comment by sohojmanush** [comment]
The pre packaged version 1.7.0 dmg crashes on macOS 10.15. But, pip install does work. Too bad it doesnt have an icon in the application folder.
